test/structural_natural_bijections.jl

```diff
@@ -0,0 +1,84 @@
+using ChainRulesCore: backing
+
+# Need to be able to compare `Tangent`s to verify correctness of definition of natural.
+Base.:(==)(t::Tangent, s::Tangent) = (backing(t) == backing(s))
```

Comment:
Can't do it like that. Because it might be missing fields etc.
If this was in ChainRules.jl I would say use `ChainRulesTestUtils.test_approx`

Reply:
> Can't do it like that. Because it might be missing fields etc.

Ah cool -- I figured I was probably at risk of missing edge cases. I doubt it's causing problems here.

> I think though you probably don't need to test equality between tangents. Just equality between results of adding them to the primal ?

Hmm yeah, we can probably get away without comparing equality on `Tangent`s. I'm reasonably confident that we need to check equality between tangents of some kind though -- whether that be the naturals or the structurals. See the first and third tests in `check_bijections`.

Reply:
I'm comparing equality between naturals now, so the `==` between `Tangent`s has been removed.